tests/unit/lms/services/canvas_api_test.py

```diff
@@ -519,3 +600,23 @@ def access_token(db_session, access_token_fields):
     access_token = OAuth2Token(**access_token_fields)
     db_session.add(access_token)
     return access_token
+
+
+@pytest.fixture
+def Schema():
+    # pylint: disable=invalid-name
+    Schema = create_autospec(RequestsResponseSchema)
+    Schema.many = False
+
+    return Schema
+
+
+@pytest.fixture
+def PaginatedSchema(Schema):
+    Schema.many = True
+
+    Schema.return_value.parse.side_effect = (
+        [f"item_{i}"] for i in range(100)
+    )  # pragma: no cover
```

Comment:
Is there something weird about setting `side_effect` to a generator?

Reply:
No it's intended behavior. It causes the side effect to return a new item each time you call it.